gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/bottomsheet/GravatarQuickEditorBottomSheet.kt

```diff
@@ -100,41 +111,103 @@ internal fun GravatarQuickEditorBottomSheet(
     }
 }
 
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 private fun GravatarModalBottomSheet(
     onDismiss: (dismissReason: GravatarQuickEditorDismissReason) -> Unit = {},
-    modalBottomSheetState: SheetState,
+    modalBottomSheetState: ModalBottomSheetState,
     content: @Composable () -> Unit,
 ) {
     val coroutineScope = rememberCoroutineScope()
 
+    LaunchedEffect(modalBottomSheetState.currentDetent) {
+        if (modalBottomSheetState.currentDetent == Hidden) {
+            onDismiss(GravatarQuickEditorDismissReason.Finished)
+        }
+    }
+
     GravatarTheme {
         ModalBottomSheet(
-            onDismissRequest = { onDismiss(GravatarQuickEditorDismissReason.Finished) },
-            sheetState = modalBottomSheetState,
-            dragHandle = { QEDragHandle() },
-            containerColor = MaterialTheme.colorScheme.surface,
-            tonalElevation = 1.dp,
-            contentWindowInsets = { WindowInsets(0, 0, 0, 0) },
+            state = modalBottomSheetState,
         ) {
-            Surface(
-                modifier = Modifier.navigationBarsPadding(),
+            Scrim(
+                modifier = Modifier.clickable { modalBottomSheetState.currentDetent = Hidden },
+                scrimColor = MaterialTheme.colorScheme.scrim.copy(alpha = 0.32f),
+            )
+            Sheet(
+                modifier = Modifier
+                    .clip(RoundedCornerShape(topStart = 28.dp, topEnd = 28.dp))
+                    .background(MaterialTheme.colorScheme.surfaceColorAtElevation(1.dp))
+                    .widthIn(max = 640.dp)
+                    .fillMaxWidth()
+                    .padding(
+                        WindowInsets.navigationBars
+                            .only(WindowInsetsSides.Vertical)
+                            .asPaddingValues(),
+                    ),
             ) {
-                Column {
-                    QETopBar(
-                        onDoneClick = {
-                            coroutineScope.launch {
-                                modalBottomSheetState.hide()
-                                onDismiss(GravatarQuickEditorDismissReason.Finished)
-                            }
-                        },
-                    )
-                    content()
+                val window = LocalModalWindow.current
+                val isDarkTheme = isSystemInDarkTheme()
+                LaunchedEffect(Unit) {
+                    window.navigationBarColor = Color.TRANSPARENT
+                    WindowInsetsControllerCompat(window, window.decorView).isAppearanceLightNavigationBars =
+                        !isDarkTheme
+                }
```

Comment:
❓ Why not just remove that line and keep the system default?

```suggestion
                LaunchedEffect(Unit) {
                    window.navigationBarColor = Color.TRANSPARENT                   
                }
```

Reply:
Because in the light mode, the navigation bar is darker than with Material3 modal - I think it looks better like this.

Reply:
TIL - 👍